crates/checkpoint-downloader/src/downloader.rs

```diff
@@ -426,75 +424,54 @@ impl ParallelCheckpointDownloaderInner {
     }
 
     /// Downloads a checkpoint with retries.
-    #[cfg(not(test))]
     async fn download_with_retry(
         client: &RetriableRpcClient,
         sequence_number: CheckpointSequenceNumber,
         config: &ParallelDownloaderConfig,
         rng: &mut StdRng,
     ) -> CheckpointEntry {
-        let mut backoff = create_backoff(rng, config);
+        let mut backoff = if cfg!(test) {
```

Comment:
nit: Maybe add a comment about the different behavior b/w prod and test for future. it's a small thing but I think we might scratch our heads one day later and wonder why this if/else branch is needed.

Reply:
Done.